src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildInfoAction.java

```diff
@@ -0,0 +1,41 @@
+package org.jenkinsci.plugins.workflow.support.steps.build;
+
+import hudson.model.InvisibleAction;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class BuildInfoAction extends InvisibleAction {
+    private final List<BuildInfo> buildInfoList = new ArrayList<>();
+
+    public BuildInfoAction(String projectName, int buildNumber) {
+        buildInfoList.add(new BuildInfo(projectName, buildNumber));
+    }
+
+    public List<BuildInfo> getBuildInfoList() {
+        return buildInfoList;
+    }
+
+    public void addBuildInfo(String projectName, int buildNumber) {
+        buildInfoList.add(new BuildInfo(projectName, buildNumber));
```

Comment:
It seems there may be concurrency issues with this field. I would recommend CopyOnWrite list or `synchronize`